jaxley/io/swc.py

```diff
@@ -88,15 +96,15 @@ def _split_into_branches(
     current_branch = []
     all_types = []
 
-    # Loop over every line in the SWC file.
-    for c in content:
+    # `previous_type` tracks the type (soma, axon...) of the previous SWC node.
+    previous_type = 0
+    for c in content:  # Loop over every line in the SWC file.
         current_ind = c[0]  # First col is row_identifier
```

Comment:
Just to make the typing consistent we could typecast the input as int here

`
current_ind = int(c[0])  # First col is row_identifier
current_parent = int(c[-1)]  # Last col is parent in SWC specification.
current_type = int(c[1])
`

Because sometimes the types being appended are float while sometimes they are int...

Reply:
Agreed, done!